### ISSUE TITLE
TextEllipsis should be able to be shown without a Tooltip

### ISSUE THREAD

**Issue by anastasialanz** [dequelabs/cauldron#1842]
[TextEllipses](https://cauldron.dequelabs.com/components/TextEllipsis) currently mandates that a Tooltip is shown and the ellipsized text becomes keyboard accessible when the text no longer fits within the established container. However there is an exception to [WCAG 1.4.12 Text Spacing](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html) in that there is an additional exemption to appearing on focus or activation:

> [...] where the ellipsis is part of a section of content which includes a link, the truncated text is revealed on the linked page

By default, we still want to have the truncated text appear by default for this component via activation on the page without any other property changes. But given that there are scenarios where we employ a link to content where the text can be revealed we want to allow for the tooltip to be hidden.

## Implementation

```ts
interface TextEllipsis {
  hideTooltip?: boolean
}
```

When this property is set to `true`, no tooltip should be shown and the text should be ellipsized with no additional keyboard access provided.

We should also provide additional documentation guidance as our goal is to provide accessible components _by default_ when possible. Setting `hideTooltip` to `false` will _require_ additional mechanisms leading towards an outside solution to address this specific accessibility problem.

> **Important:** This prop should only be used in specific circumstances that comply with additional WCAG guidelines. Setting `hideTooltip={true}` should **only** be used when the full non-truncated text is available via a linked page. Incorrect usage of this property will result in an accessibility issue
